src/main/clojure/clojure/core/matrix.clj

```diff
@@ -1701,6 +1701,119 @@
   ([m]
     (mp/element-max m)))
 
+(defn clamp
+  "Clamps each element in a numerical array between an upper and lower bound.
+  Returns an error if the lower bound is greater that the upper bound.
```

Comment:
Yeah I think better to leave unspecified. Testing for special cases and throwing exceptions can expensive, some implementations will not want to do it.